ls1intum/Artemis
# `Development`: Improve Atlassian setup script

### Checklist
#### General



### Motivation and Context


When setting up the Atlassian toolchain (Jira, Bitbucket, Bamboo) for development locally, the already existing script `atlassian-setup.sh` creates the users and groups automatically.
However, the assignment of the created users to their respective group in Jira currently needs to be performed manually, which can be error prone and cumbersome.
In addition, when the script is executed on MacOS, it throws errors, because xdc-utils is not installed by default.

### Description

This change reenables (uncomment) the automatic assignment of users to their respective group.
In addition, the application link configuration pages now open in the default browser for MacOS users instead of throwing an error.

### Steps for Testing

Prerequisites:
- Remove users (artemis_test_user_{1-20}) and groups (students, instructors, editors, tutors) from Jira
- Delete all application links in Jira, Bamboo and Bitbucket

1. Run the script `atlassian-setup.sh`
(For MacOS users): After the script is executed, three tabs pointing to the application links configuration page of the respective Atlassian system should be opened in your default browser.
2. Make sure the users are created in JIRA and are assigned to their corresponding group.

| group | user |
| ------------- | ------------- |
| students  | artemis_test_user_{1-5}  |
| tutors  | artemis_test_user_{6-10}  |
| editors  | artemis_test_user_{11-15}  |
| instructors  | artemis_test_user_{16-20}  |
3. Also remember to change the display url for the links Bitbucket -> Bamboo and Bamboo -> Bitbucket back to localhost according to step 4 in the [setup guide](https://docs.artemis.ase.in.tum.de/dev/setup/bamboo-bitbucket-jira/).


### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


### Screenshots
